vital/util/buffer.h

```diff
@@ -0,0 +1,56 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief Base interface for simple buffer classes
```

Comment:
I suggest we remove all of this buffer / ring buffer stuff.

Reply:
I agree.